src/test/java/org/cactoos/map/SolidMapTest.java

```diff
@@ -81,4 +86,71 @@ public void mapsToSameObjects() throws Exception {
             map.get(0), Matchers.equalTo(map.get(0))
         );
     }
+
+    @Test
+    @SuppressWarnings({"unchecked", "MismatchedQueryAndUpdateOfCollection"})
+    public void extendsExistingMapWithArrayOfEntries() {
+        final SolidMap<Integer, Integer> map = new SolidMap<>(
+            new MapEntry<>(0, 10),
+            new MapEntry<>(1, 11)
+        );
+        final SolidMap<Integer, Integer> toadd = new SolidMap<>(
+            new MapEntry<>(2, 12),
+            new MapEntry<>(3, 13)
+        );
+        final Map.Entry<Integer, Integer>[] array =
+            (Map.Entry<Integer, Integer>[]) new Map.Entry<?, ?>[toadd.size()];
+        toadd.entrySet().toArray(array);
+        MatcherAssert.assertThat(
+            "Does not extends existing map with array of entries",
+            new SolidMap<Integer, Integer>(map, array).size(),
```

Comment:
@flslkxtc This ctor accepts a ``vararg`` array (``...``) -- isn't it possible to instantiate ``Map.Entry`` on the spot and remove ``toadd`` and ``array``? 

At least, if this is not possible, get rid of ``toadd`` and just say 

```java
final Map.Entry<Integer, Integer>[] array =
    (Map.Entry<Integer, Integer>[]) new Map.Entry<?, ?>[toadd.size()];
new SolidMap<>(
    new MapEntry<>(2, 12),
   new MapEntry<>(3, 13)
).entrySet().toArray(
    array
);
```